src/apps/service/apps/de/jobs/io_tickets.clj

```diff
@@ -0,0 +1,42 @@
+(ns apps.service.apps.de.jobs.io-tickets
+  (:require [apps.clients.data-info :as data-info]
+            [apps.service.apps.jobs.util :as util]))
+
+(defn update-with-download-tickets
+  "Takes a list if input params,
+   creates a download ticket for the path of each input's :value,
+   and updates those inputs with the created ticket in a :ticket field.
+   For use with apps.service.apps.de.jobs.protocol/buildInputs"
+  [user params]
+  (if (or (empty? params) (every? (comp empty? :value) params))
+    params
+    (let [paths      (remove empty? (map :value params))
+          ticket-map (->> (data-info/create-tickets user
+                                                    paths
+                                                    :mode       "read"
+                                                    :public     false
+                                                    :uses-limit 1)
+                          :tickets
+                          (map #(vector (:path %) (:ticket-id %)))
```

Comment:
I like to use `juxt` for stuff like this:

```clojure
(map (juxt :path :ticket-id))
```

Reply:
Yes, that's a much better pattern. Thanks for the reminder 👍